tools/clang/lib/SPIRV/EmitSPIRVAction.cpp

```diff
@@ -23,25 +23,90 @@
 
 namespace clang {
 namespace {
-class SPIRVEmitter : public ASTConsumer,
-                     public RecursiveASTVisitor<SPIRVEmitter> {
+
+class SPIRVEmitter : public ASTConsumer {
 public:
   explicit SPIRVEmitter(raw_ostream *Out)
-      : OutStream(*Out), TheContext(), Builder(&TheContext) {}
+      : OutStream(*Out), TheContext(), TheBuilder(&TheContext) {}
 
   void HandleTranslationUnit(ASTContext &Context) override {
-    Builder.beginModule();
-    Builder.endModule();
-    std::vector<uint32_t> M = Builder.takeModule();
+    TheBuilder.requireCapability(spv::Capability::Shader);
```

Comment:
Hmm, we won't target OpenCL kernels. So we don't need `OpCapability Kernel`. Note that `GLCompute` requires `Shader` capability. https://www.khronos.org/registry/spir-v/specs/1.0/SPIRV.html#Execution_Model